code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/snapshot/SnapshotFilter.java

```diff
@@ -0,0 +1,81 @@
+package io.cattle.platform.iaas.api.filter.snapshot;
+
+import io.cattle.platform.core.constants.SnapshotConstants;
+import io.cattle.platform.core.model.Snapshot;
+import io.cattle.platform.core.model.Volume;
+import io.cattle.platform.iaas.api.filter.common.AbstractDefaultResourceManagerFilter;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.util.DataUtils;
+import io.github.ibuildthecloud.gdapi.exception.ValidationErrorException;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManager;
+import io.github.ibuildthecloud.gdapi.validation.ValidationErrorCodes;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SnapshotFilter extends AbstractDefaultResourceManagerFilter {
+
+    private static final Logger log = LoggerFactory
+            .getLogger(SnapshotFilter.class);
+    ObjectManager objectManager;
+
+    @Override
+    public String[] getTypes() {
+        return new String[] { "snapshot" };
+    }
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[] { Snapshot.class };
+    }
+
+    @Inject
+    public void setObjectManager(ObjectManager objectManager) {
+        this.objectManager = objectManager;
+    }
+
+    @Override
+    public Object create(String type, ApiRequest request, ResourceManager next) {
+        Long volumeId = DataUtils.getFieldFromRequest(request,
```

Comment:
Don't need to check for null volume if volumeId is set to required in the API schema